source/common/config/utility.h

```diff
@@ -108,6 +119,10 @@ class Utility {
   static SubscriptionStats generateStats(Stats::Scope& scope) {
     return {ALL_SUBSCRIPTION_STATS(POOL_COUNTER(scope))};
   }
+
+  static const std::string REST_LEGACY;
```

Comment:
nit: This should use the const singleton pattern. See Http::Headers

Reply:
Sorry, I am confused about this. If declare and initialize like in `source/common/http/headers.h` with curly brace initialization the compiler complains about `in-class initialization of static data member of non-literal type`, but of course I can't make them non-static because I would be using the variable in a static member function. What am I missing?

Reply:
You create a class with non-static members, and then access it via the static `get()` in https://github.com/lyft/envoy/blob/master/source/common/common/singleton.h.

Reply:
ah gotcha, that pattern!

Reply:
Nice this makes it better. I knew I had seen this as the Const counter part to the mutable singleton, should have scrolled to the end of the file. Another tool in the toolshed 👍